stac_fastapi/core/stac_fastapi/core/core.py

```diff
@@ -743,6 +747,21 @@ async def post_search(
         Raises:
             HTTPException: If there is an error with the cql2_json filter.
         """
+        global_max_limit = int(os.getenv("STAC_GLOBAL_ITEM_MAX_LIMIT", 100))
```

Comment:
concerned here too about setting a default. It may work for some Users but maybe not others.